src/readers/morphologySWC.cpp

```diff
@@ -14,30 +15,49 @@
 #include <morphio/properties.h>
 
 namespace {
+// It's not clear if -1 is the only way of identifying the root section.
+const int SWC_UNDEFINED_PARENT = -1;
+const unsigned int SWC_ROOT = 0xFFFFFFFD;
+
 bool _ignoreLine(const std::string& line) {
     std::size_t pos = line.find_first_not_of("\n\r\t ");
     return pos == std::string::npos || line[pos] == '#';
 }
 
 morphio::readers::Sample readSample(const char* line, unsigned int lineNumber_) {
 #ifdef MORPHIO_USE_DOUBLE
-    const char* const format = "%u%d%lg%lg%lg%lg%d";
+    const char* const format = "%d%d%lg%lg%lg%lg%d";
 #else
-    const char* const format = "%u%d%f%f%f%f%d";
+    const char* const format = "%d%d%f%f%f%f%d";
 #endif
 
     morphio::floatType radius = -1.;
     int int_type = -1;
     morphio::readers::Sample sample;
+    int parentId = -1;
+    int id = -1;
     sample.valid = sscanf(line,
                           format,
-                          &sample.id,
+                          &id,
                           &int_type,
                           &sample.point[0],
                           &sample.point[1],
                           &sample.point[2],
                           &radius,
-                          &sample.parentId) == 7;
+                          &parentId) == 7;
```

Comment:
Maybe turn 7 into a constant?

Reply:
this will be going away once I get rid of `scanf`